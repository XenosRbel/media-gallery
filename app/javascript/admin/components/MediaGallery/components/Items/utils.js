export const TRANSLATE_RENAME = I18n.t('admin.javascript.media_gallery.folder.rename');
export const TRANSLATE_DELETE = I18n.t('admin.javascript.media_gallery.folder.delete');
export const TRANSLATE_CONFIRM_ATTRIBUTE = I18n.t('admin.sure?');
