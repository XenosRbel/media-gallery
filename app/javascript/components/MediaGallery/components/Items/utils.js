export const TRANSLATE_RENAME = 'Переименовать'
export const TRANSLATE_DELETE = 'Удалить'
export const TRANSLATE_CONFIRM_ATTRIBUTE = 'Вы уверены?'
