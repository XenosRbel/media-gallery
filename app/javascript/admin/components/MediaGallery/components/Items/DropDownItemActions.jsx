import React, { useMemo, useCallback, useContext } from 'react';
import PropTypes from 'prop-types';
import { Menu, Dropdown } from 'antd';

import { TRANSLATE_RENAME, TRANSLATE_DELETE, TRANSLATE_CONFIRM_ATTRIBUTE } from './utils';
import { DELETE_HTTP_METHOD, DEFAULT_SYSTEM_GROUP_NAME } from '../../constants/common';
import { submitButton } from '../../../common/utils';
import { MediaGalleryContext } from '../MediaGalleryContext';

const DropDownItemActions = ({ folderAttr, links: { folderPath } }) => {
  const { handleError, handleLoadGalleryItems, handleRenameGroup } = useContext(MediaGalleryContext);

  const handleDeleteGroup = useCallback(async () => {
    const data = await submitButton(DELETE_HTTP_METHOD, folderPath);

    handleLoadGalleryItems();
    handleError(data);
  }, [folderPath, handleError, handleLoadGalleryItems]);

  const items = useMemo(
    () => (
      <menu>
        <li>
          <a onClick={() => handleRenameGroup({ folder: folderAttr, path: folderPath })}>
            <span>{TRANSLATE_RENAME}</span>
          </a>
        </li>
        <li>
          <a data-confirm={TRANSLATE_CONFIRM_ATTRIBUTE} onClick={handleDeleteGroup}>
            <span>{TRANSLATE_DELETE}</span>
          </a>
        </li>
      </menu>
    ),
    [folderAttr, folderPath, handleDeleteGroup, handleRenameGroup],
  );

  return (
    <Dropdown
      menu={items}
      disabled={folderAttr.name === DEFAULT_SYSTEM_GROUP_NAME}
      onClick={(event) => {
        event.stopPropagation();
      }}
    >
      <a className="ant-dropdown-link ant-btn ant-btn-link ant-btn-icon-only">
        <i className="fa fa-ellipsis-v" />
      </a>
  </Dropdown>
  );
};

DropDownItemActions.propTypes = {
  folderAttr: PropTypes.shape({
    name: PropTypes.string.isRequired,
    systemName: PropTypes.string.isRequired,
  }),
  links: PropTypes.shape({
    folderPath: PropTypes.string,
  }).isRequired,
};

DropDownItemActions.defaultProps = {
  folderAttr: {
    name: '',
    systemName: '',
  },
};

export default DropDownItemActions;
