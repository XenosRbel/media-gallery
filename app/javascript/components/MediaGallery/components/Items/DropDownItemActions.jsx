import React, { useMemo, useCallback, useContext } from 'react';
import PropTypes from 'prop-types';
import { Menu, Dropdown } from 'antd';

import { TRANSLATE_RENAME, TRANSLATE_DELETE, TRANSLATE_CONFIRM_ATTRIBUTE } from './utils';
import { DELETE_HTTP_METHOD, DEFAULT_SYSTEM_GROUP_NAME } from '../../constants/common';
import { submitButton } from '../../../common/utils';
import { MediaGalleryContext } from '../MediaGalleryContext';

const DropDownItemActions = ({ groupAttr, links: { groupPath } }) => {
  const { handleError, handleLoadGalleryItems, handleRenameGroup } = useContext(MediaGalleryContext);

  const handleDeleteGroup = useCallback(async () => {
    const data = await submitButton(DELETE_HTTP_METHOD, groupPath);

    handleLoadGalleryItems();
    handleError(data);
  }, [groupPath, handleError, handleLoadGalleryItems]);

  const menu = useMemo(
    () => (
      <Menu>
        <Menu.Item>
          <a onClick={() => handleRenameGroup({ group: groupAttr, path: groupPath })}>
            <span>{TRANSLATE_RENAME}</span>
          </a>
        </Menu.Item>
        <Menu.Item>
          <a data-confirm={TRANSLATE_CONFIRM_ATTRIBUTE} onClick={handleDeleteGroup}>
            <span>{TRANSLATE_DELETE}</span>
          </a>
        </Menu.Item>
      </Menu>
    ),
    [groupAttr, groupPath, handleDeleteGroup, handleRenameGroup],
  );

  return (
    <>
      <Dropdown
        overlay={menu}
        disabled={groupAttr.name === DEFAULT_SYSTEM_GROUP_NAME}
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <a className="antd-dropdown-link antd-btn antd-btn-link antd-btn-icon-only">
          <i className="fa fa-ellipsis-v" />
        </a>
      </Dropdown>
    </>
  );
};

DropDownItemActions.propTypes = {
  groupAttr: PropTypes.shape({
    name: PropTypes.string.isRequired,
    systemName: PropTypes.string.isRequired,
  }),
  links: PropTypes.shape({
    groupPath: PropTypes.string,
  }).isRequired,
};

DropDownItemActions.defaultProps = {
  groupAttr: {
    name: '',
    systemName: '',
  },
};

export default DropDownItemActions;
