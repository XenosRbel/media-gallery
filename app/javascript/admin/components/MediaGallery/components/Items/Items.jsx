import React, { cloneElement } from 'react';
import PropTypes from 'prop-types';
import { Collapse } from 'antd';

import DropDownItemActions from './DropDownItemActions';
import { DEFAULT_ACTIVE_ACCORDION_KEY } from '../../constants/items';

const Items = ({ records, shouldShowGroupActions, galleryComponent }) => (
  <Collapse defaultActiveKey={DEFAULT_ACTIVE_ACCORDION_KEY}>
    {records.map(({ id, attributes: { name, systemName, files }, links }) => (
      <Collapse.Panel
        header={<span className="items-collapse-header">{name}</span>}
        key={id}
        extra={shouldShowGroupActions && <DropDownItemActions links={links} folderAttr={{ name, systemName }} />}
      >
        {cloneElement(galleryComponent, { files, folderId: id, folderSystemName: systemName })}
      </Collapse.Panel>
    ))}
  </Collapse>
);

Items.propTypes = {
  galleryComponent: PropTypes.node.isRequired,
  records: PropTypes.arrayOf(
    PropTypes.shape({
      attributes: PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        systemName: PropTypes.string.isRequired,
      }),
      id: PropTypes.string.isRequired,
    }).isRequired,
  ),
  shouldShowGroupActions: PropTypes.bool,
};

Items.defaultProps = {
  records: [],
  shouldShowGroupActions: true,
};

export default Items;
