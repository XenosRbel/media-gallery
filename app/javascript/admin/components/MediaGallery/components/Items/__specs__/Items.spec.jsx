import React from 'react';
import { shallow } from 'enzyme';

import Items from '../Items';

describe('Component: Items', () => {
  const galleryComponent = <div> </div>;

  const wrapper = shallow(<Items galleryComponent={galleryComponent} />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
