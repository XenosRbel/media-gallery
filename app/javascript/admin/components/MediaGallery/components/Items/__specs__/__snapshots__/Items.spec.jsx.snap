// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Items should render correctly 1`] = `
<q
  bordered={true}
  defaultActiveKey={
    Array [
      "1",
    ]
  }
/>
`;
